renderers/css/actions.py

```diff
@@ -36,13 +36,16 @@ class OpiExecuteCommand(OpiAction):
 class OpiOpen(OpiAction):
     """Renderer for write PV actions."""
     ACTION_TYPE = 'OPEN_DISPLAY'
+    MACRO_ERROR = 'Macros must have key and value as strings ({}:{} violates this rule)'
 
     def render(self, actions_node, action_model):
         action_node = super(OpiOpen, self).render(actions_node, action_model)
         macros_node = et.SubElement(action_node, 'macros')
         parent_macros_node = et.SubElement(macros_node, 'include_parent_macros')
         parent_macros_node.text = 'true' if action_model._parent_macros else 'false'
         for key, value in action_model._macros.items():
+            if not utils.is_string(key) or not utils.is_string(value):
```

Comment:
Good points.

1.  I didn't think of checking for valid XML tags.  We get the error `ValueError: Invalid tag name u'1'`, which we could wrap and improve, but is not so bad.
2.  You could define a class that represents your macros and renders itself on `__str__`, but is that really a use-case?  I wanted to avoid things like 1 getting rendered as 1.0.